src/Symfony/Component/HttpClient/Internal/LegacyHttplugInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Internal;
+
+use Http\Client\HttpClient;
+use Http\Message\RequestFactory;
+use Http\Message\StreamFactory;
+use Http\Message\UriFactory;
+
+if (interface_exists(RequestFactory::class)) {
```

Comment:
What about the existence of `HttpClient` ?

Reply:
It's always here, because we check for HttpAsyncClient, which covers HttpClient
I realized I forgot to HttpClient on the "else" path, updated.